Shared/Samples/Edit geometries with programmatic reticle tool/EditGeometriesWithProgrammaticReticleToolView.swift

```diff
@@ -0,0 +1,599 @@
+            model.geometryEditor.clearGeometry()
+        } label: {
+            Label("Clear Current Sketch", systemImage: "trash")
+        }
+        .disabled(!canClearCurrentSketch)
+        
+        Divider()
+        
+        Button {
+            model.save()
+        } label: {
+            Label("Save Sketch", systemImage: "square.and.arrow.down")
+        }
+        .disabled(!canSave)
+        
+        Button {
+            model.stop()
+        } label: {
+            Label("Cancel Sketch", systemImage: "xmark")
+        }
+    }
+}
+
+private extension GeometryEditorMenu {
+    /// A Boolean value indicating whether the selection can be deleted.
+    ///
+    /// In some instances deleting the selection may be invalid. One example would be the mid vertex
+    /// of a line.
+    var deleteButtonIsDisabled: Bool {
+        guard let selectedElement else { return true }
+        return !selectedElement.canBeDeleted
+    }
+    
+    /// A Boolean value indicating if the geometry can be saved to a graphics overlay.
+    var canSave: Bool {
+        return geometry?.sketchIsValid ?? false
+    }
+    
+    /// A Boolean value indicating if the geometry can be cleared from the geometry editor.
+    var canClearCurrentSketch: Bool {
+        return geometry.map { !$0.isEmpty } ?? false
+    }
+}
+
+/// An object that acts as a view model for the geometry editor menu.
+@MainActor
+@Observable
+private class GeometryEditorModel {
+    /// The geometry editor.
+    let geometryEditor = GeometryEditor()
+    /// The programmatic reticle tool.
+    private let reticleTool = ProgrammaticReticleTool()
+    /// The graphics overlay used to save geometries to.
+    let geometryOverlay = GraphicsOverlay(renderingMode: .dynamic)
+    /// The selected graphic to edit.
+    @ObservationIgnored var selectedGraphic: Graphic?
+    /// A Boolean value indicating if the initial graphics and saved sketches can be cleared.
+    private(set) var canClearGraphics = true
+    /// A Boolean value indicating if the geometry editor has started.
+    private(set) var isStarted = false
+    /// A Boolean value indicating if vertex creation is allowed.
+    var allowsVertexCreation = true {
+        didSet {
+            reticleTool.vertexCreationPreviewIsEnabled = allowsVertexCreation
+            reticleTool.style.growEffect?.appliesToMidVertices = allowsVertexCreation
+        }
+    }
+    /// The reticle state used to determine the current action of the programmatic reticle tool.
+    enum ReticleState {
+        case `default`, pickedUp, hoveringVertex, hoveringMidVertex
+        
+        /// A human-readable label of the property.
+        var label: String {
+            switch self {
+            case .default: "Insert Point"
+            case .pickedUp: "Drop Point"
+            case .hoveringVertex: "Pick up point"
+            case .hoveringMidVertex: "Pick up point"
+            }
+        }
+    }
+    /// The programmatic reticle state.
```

Comment:
We offset type declarations from other declarations with a blank line.

```suggestion
    }
    
    /// The reticle state used to determine the current action of the programmatic reticle tool.
    enum ReticleState {
        case `default`, pickedUp, hoveringVertex, hoveringMidVertex
        
        /// A human-readable label of the property.
        var label: String {
            switch self {
            case .default: "Insert Point"
            case .pickedUp: "Drop Point"
            case .hoveringVertex: "Pick up point"
            case .hoveringMidVertex: "Pick up point"
            }
        }
    }
    
    /// The programmatic reticle state.
```